### PR TITLE
allows PrepareTestContainer to be used by other projects

### PR DESCRIPTION
**Pull request by Lexman42**

### CODE REVIEW

**Comment by catsby** [comment]
Hello! I am generally shy of exporting methods like this, could you share the use case(s)? Would it be better to extract this to https://github.com/hashicorp/vault/blob/fd1542d0485b99e4dddd3aee265a46f969e6a0c4/helper/testhelpers/docker/testhelpers.go albeit more generically,  and maybe some methods there too to easily mount backends? In this case, there is very little that's actually specific to Transit:

```
	// Set up transit
		if err := vault.Sys().Mount(testMountPath, &api.MountInput{
			Type: "transit",
		}); err != nil {
			return err
		}

		// Create default aesgcm key
		if _, err := vault.Logical().Write(path.Join(testMountPath, "keys", testKeyName), map[string]interface{}{}); err != nil {
			return err
		}
```

`prepareTestContainer` appears to have different versions in the codebase, likely all very similar:

```bash
[vault][master](4)$ ack "func prepareTestContainer\(" | grep -v vendor 
physical/postgresql/postgresql_test.go:364:func prepareTestContainer(t *testing.T, logger log.Logger) (cleanup func(), retConnString string) {
builtin/logical/postgresql/backend_test.go:22:func prepareTestContainer(t *testing.T) (cleanup func(), retURL string) {
builtin/logical/nomad/backend_test.go:19:func prepareTestContainer(t *testing.T) (cleanup func(), retAddress string, nomadToken string) {
builtin/logical/mysql/backend_test.go:19:func prepareTestContainer(t *testing.T) (func(), string) {
vault/seal/transit/transit_acc_test.go:116:func prepareTestContainer(t *testing.T) (cleanup func(), retAddress, token, mountPath, keyName string, tlsConfig *api.TLSConfig) {

```

**Comment by Lexman42** [comment]
Hello,

The use case is extending the transit seal via it's kms-wrapper and wanting to test that extended functionality.

Agreed that moving this to the docker package as you have suggested makes more sense. I will update the PR to do that instead. After doing so, the other places that you have identified that use a similar `prepareTestContainer` could probably leverage the one in the docker container and as they do, the docker.prepareTestContainer can be generecized to work for each similar use case.

File: helper/testhelpers/docker/testhelpers.go [hunk 1]
```diff
@@ -23,3 +27,75 @@ func CleanupResource(t testing.T, pool *dockertest.Pool, resource *dockertest.Re
 	}
 	t.Fatalf("Failed to cleanup local container: %s", err)
 }
+
+func PrepareTestContainer(t *testing.T) (cleanup func(), retAddress, token, mountPath, keyName string, tlsConfig *api.TLSConfig) {
```
Comment (ncabatoff) [inline helper/testhelpers/docker/testhelpers.go]: This func name is pretty generic.  Usually it's used in a package whose name reflects what kind of container we're starting (e.g. consul.PrepareTestContainer) but here the package is `docker`, which doesn't help.  Maybe `PrepareVaultContainer`?

File: helper/testhelpers/docker/testhelpers.go [hunk 2]
```diff
@@ -23,3 +27,75 @@ func CleanupResource(t testing.T, pool *dockertest.Pool, resource *dockertest.Re
...
+			return err
+		}
+		vault, err := api.NewClient(vaultConfig)
+		if err != nil {
+			return err
+		}
+		vault.SetToken(testToken)
+
+		// Set up transit
+		if err := vault.Sys().Mount(testMountPath, &api.MountInput{
```
Comment (ncabatoff) [inline helper/testhelpers/docker/testhelpers.go]: I feel this doesn't belong in a generic helper.  Why not move mounting of transit and creating of keys within to the test that cares about them?

File: helper/testhelpers/docker/testhelpers.go (hunk 1 above)
Comment (Lexman42) [reply helper/testhelpers/docker/testhelpers.go]: changed to `PrepareVaultContainer`

File: helper/testhelpers/docker/testhelpers.go (hunk 2 above)
Comment (Lexman42) [reply helper/testhelpers/docker/testhelpers.go]: Agreed, I moved setting up Transit to its own function. Both of the functions here are still pretty transit specific, my thinking is that we can generecize as tests begin to leverage this package. The other is to make it super generic from the get-go.

**Comment by Lexman42** [comment]
didn't go this route after all.

**Pull request closed without merging**